cli/azd/pkg/environment/manager.go

```diff
@@ -74,8 +75,17 @@ func NewManager(
 	// via the container but we can't do that because the remote data store is optional and the IoC
 	// container doesn't support optional interface based dependencies.
 	if remoteConfig != nil {
-		if err := serviceLocator.ResolveNamed(remoteConfig.Backend, &remote); err != nil {
-			return nil, fmt.Errorf("resolving remote data store: %w", err)
+		err := serviceLocator.ResolveNamed(remoteConfig.Backend, &remote)
+		if err != nil {
+			if errors.Is(err, ioc.ErrResolveInstance) {
+				return nil, fmt.Errorf(
+					"remote state configuration is invalid. The specified backend '%s' is not valid. Valid values are '%s'.",
+					remoteConfig.Backend,
+					strings.Join(ValidRemoteKinds, ","),
```

Comment:
nit. You can use `ListAsText` from ux package. Like

```suggestion
					ux.ListAsText(ValidRemoteKinds),
```

I added this function for having coma and/or `and` as the join token (when needed) :)